### PR TITLE
Update elvish to 0.18.0

### PR DESCRIPTION
**Pull request by skwerlman**
# Summary

This PR replaces the newly deprecated 'except' with 'catch' to fix deprecation warnings in elvish 0.18.

This is a breaking change, as it drops support for elvish <0.18

Fixes: N/A

## Other Information

Upstream change: elves/elvish#1497

### CODE REVIEW

**Comment by Stratus3D** [comment]
Can anyone who uses elvish test this for me? Code changes look good to me.

**Review by Stratus3D — APPROVED** [review]

**Comment by Stratus3D** [comment]
This PR failed the semantic commit lint, so we need to correct the commit message before merging.

**Comment by skwerlman** [comment]
> This PR failed the semantic commit lint, so we need to correct the commit message before merging.

happy to correct the message to whatever is needed, but it seems to me that the lint didnt actually output what is wrong with the message. is there a guide i missed somewhere for correct commit message formatting?

**Comment by Stratus3D** [comment]
```
Error: No release type found in pull request title "Update elvish to 0.18.0". Add a prefix to indicate what kind of release this pull request corresponds to (see https://www.conventionalcommits.org/).

Available types:
 - feat: A new feature
 - fix: A bug fix
 - docs: Documentation only changes
 - style: Changes that do not affect the meaning of the code (white-space, formatting, missing semi-colons, etc)
 - refactor: A code change that neither fixes a bug nor adds a feature
 - perf: A code change that improves performance
 - test: Adding missing tests or correcting existing tests
 - build: Changes that affect the build system or external dependencies (example scopes: gulp, broccoli, npm)
 - ci: Changes to our CI configuration files and scripts (example scopes: Travis, Circle, BrowserStack, SauceLabs)
 - chore: Other changes that don't modify src or test files
 - revert: Reverts a previous commit
```

So instead of `Update elvish to 0.18.0` for the commit title, something like `fix: update elvish to 0.18.0` is what should be used.

**Comment by skwerlman** [comment]
i went with `feat:` for consistency with the 0.17.0 update commit

**Comment by Stratus3D** [comment]
Sorry, if we go with `feat` we can't merge this as `feat` will require a minor version bump to be released. We just tagged `0.10.0` a few days ago so I only want to release a patch version next.

**Comment by Stratus3D** [comment]
Changes have been merged in this commit - https://github.com/asdf-vm/asdf/commit/5a89563c0a37f244fa3daa46c5100b7711edde1d

**Pull request closed without merging**
